packages/frontend/src/components/AddAppConnection/index.tsx

```diff
@@ -107,68 +112,85 @@ export default function AddAppConnection(
-          : formatMessage('app.addConnection')}
-      </DialogTitle>
-
-      {authDocUrl && (
-        <Alert severity="info" sx={{ fontWeight: 300 }}>
-          {formatMessage('addAppConnection.callToDocs', {
-            appName: name,
-            docsLink: generateExternalLink(authDocUrl),
-          })}
-        </Alert>
-      )}
-
-      {error && (
-        <Alert
-          severity="error"
-          sx={{ mt: 1, fontWeight: 500, wordBreak: 'break-all' }}
-        >
-          <>
-            {error.message}
-            {error.details && (
-              <pre style={{ whiteSpace: 'pre-wrap' }}>
-                {JSON.stringify(error.details, null, 2)}
-              </pre>
-            )}
-          </>
-        </Alert>
-      )}
-
-      <DialogContent>
-        <DialogContentText tabIndex={-1} component="div">
+    <Modal
+      isOpen={true}
+      onClose={() => onClose({})}
+      data-test="add-app-connection-dialog"
+    >
+      <ModalOverlay />
+      <ModalContent>
+        <ModalHeader>
+          {hasConnection ? 'Edit connection' : 'Add connection'}
+        </ModalHeader>
+        <ModalCloseButton />
+
+        {authDocUrl && (
+          <Alert status="info" fontWeight="300" px={8}>
+            <AlertIcon />
+            Visit our
+            <Link isExternal href={authDocUrl} target="_blank" mx={1}>
+              guide
+            </Link>
+            to learn more about how to connect {name}.
+          </Alert>
+        )}
+
+        {error && (
+          <Alert
+            status="error"
+            mt={1}
+            fontWeight="500"
+            wordBreak="break-all"
+            px={8}
+          >
+            <AlertIcon />
+            <>
+              {error.message}
+              {error.details && (
+                <pre style={{ whiteSpace: 'pre-wrap' }}>
+                  {JSON.stringify(error.details, null, 2)}
+                </pre>
+              )}
+            </>
+          </Alert>
+        )}
+
+        <ModalBody>
           <Form onSubmit={submitHandler}>
-            {auth?.fields?.map((field: IField) => (
-              <InputCreator key={field.key} schema={field} />
-            ))}
-
-            <LoadingButton
-              type="submit"
-              variant="contained"
-              color="primary"
-              sx={{ boxShadow: 2 }}
-              loading={inProgress}
-              data-test="create-connection-button"
-            >
-              {formatMessage('addAppConnection.submit')}
-            </LoadingButton>
+            <VStack gap={2} pt={4} pb={8} alignItems="flex-start">
+              {auth?.fields?.map((field: IField) => (
+                <InputCreator key={field.key} schema={field} />
+              ))}
+
+              <Button
+                type="submit"
+                variant="solid"
+                colorScheme="primary"
+                isLoading={inProgress}
+                data-test="create-connection-button"
+              >
```

Comment:
nit: can consider adding `isFullWidth` to align with the add new connection modal
```suggestion
                isFullWidth
```